src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -175,38 +185,26 @@ class TezosNodeOperator(val node: TezosRPCInterface, batchConf: BatchFetchConfig
           logger.info("No new blocks to fetch from the network")
           Future.successful(List.empty)
         } else {
-          if (maxLevel == -1) logger.warn("There were apparently no blocks in the database. Downloading the whole chain..")
+          if (maxLevel == -1) logger.warn("There were apparently no blocks in the database. Downloading the whole chain.")
           else logger.info("Found new block head at level {}, currently stored max is {}. Fetching missing blocks", headLevel, maxLevel)
           getBlocks(network, headHash, maxLevel + 1, headLevel)
         }
     } yield blocks
 
-  /**
-    * Gets all blocks.
-    * @param network    Which Tezos network to go against
-    * @return           Blocks and Account hashes involved
-    */
-  def getAllBlocks(network: String): Future[List[(Block, List[AccountId])]] =
-    for {
-      blockHead <- getBlockHead(network)
-      headLevel = blockHead.metadata.header.level
-      headHash = blockHead.metadata.hash
-      blocks <- getBlocks(network, headHash, 1, headLevel)
-    } yield blocks
-
   /**
     * Gets last n blocks.
     * @param network    Which Tezos network to go against
-    * @param count      Number of latest block to fetch
+    * @param depth      Number of latest block to fetch, `None` to get all blocks
     * @return           Blocks and Account hashes involved
     */
-  def getLatestBlocks(network: String, count: Int): Future[List[(Block, List[AccountId])]] =
-    for {
-      blockHead <- getBlockHead(network)
-      headLevel = blockHead.metadata.header.level
-      headHash = blockHead.metadata.hash
-      blocks <- getBlocks(network, headHash, max(1, headLevel - count + 1), headLevel)
-    } yield blocks
+  def getLatestBlocks(network: String, depth: Option[Int] = None): Future[List[(Block, List[AccountId])]] =
+    getBlockHead(network).flatMap {
```

Comment:
Could you explain the nature of the change here?

Reply:
I merged two operations was concrete difference was only in a default parameter value.

While java encourages to overload methods with different signatures, it's common in scala to use default parameters to avoid repetition.

In this specific case, I expect `getLatestBlocks` and `getAllBlocks` to essentially do the same thing, while one is limited in the result size. So I opted to put the logic in one place, to avoid duplication whenever the logic changes.